### PR TITLE
dpu: lldb: print only NR_TASKLETS thread

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lldb/source/API/SBSymbol.cpp [hunk 1]
```diff
@@ -153,6 +153,14 @@ lldb_private::Symbol *SBSymbol::get() { return m_opaque_ptr; }
 
 void SBSymbol::reset(lldb_private::Symbol *symbol) { m_opaque_ptr = symbol; }
 
+SBAddress SBSymbol::GetRawAddress() {
+  LLDB_RECORD_METHOD_NO_ARGS(lldb::SBAddress, SBSymbol, GetRawAddress);
+
+  SBAddress addr;
+  addr.SetAddress(&m_opaque_ptr->GetAddressRef());
```
Comment (JordiChauzi) [inline lldb/source/API/SBSymbol.cpp]: Could you have used `m_opaque_ptr->GetIntegerValue` to directly fetch the integer value?

Comment (ghost) [reply lldb/source/API/SBSymbol.cpp]: Indeed, and I renamed GetRawAddress into GetIntegerValue.

**Review by JordiChauzi — APPROVED** [review]

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp [hunk 2]
```diff
@@ -124,7 +123,15 @@ bool Dpu::LoadElf(const FileSpec &elf_file_path) {
   if (status != DPU_OK)
     return false;
 
-  if (!SetPrintfSequenceAddrsFromRuntimeInfo())
+  dpu_program_t *runtime = dpu_get_program(m_dpu);
+  if (!runtime)
+    return false;
+
+  int nr_threads_enabled = runtime->nr_threads_enabled;
+  if (nr_threads_enabled != -1)
```
Comment (vpalatin) [inline lldb/source/Plugins/Process/Dpu/Dpu.cpp:131]: if you have garbage in the environment variable, the strtoll will set this to 0 ?
you might just want to skip 0 either.

**Review by vpalatin — APPROVED** [review]

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp (hunk 2 above)
Comment (ghost) [reply lldb/source/Plugins/Process/Dpu/Dpu.cpp:131]: I don't expect the environment variable to be set by anything else than ``dpu_commands.py``. The variable is indeed not than easy to find, in order to make sure than no one set it by mistake.

Both ``dpu_commands.py`` and the value coming from ``runtime->nr_threads_enabled`` have their default value set to ``-1``.

``runtime->nr_threads_enabled`` is not set by a ``stroll``, but by the value read in the elf.

**Pull request merged**